crates/swc_ecma_codegen/src/lib.rs

```diff
@@ -579,7 +585,14 @@ where
             }
         }
 
-        let value = get_quoted_utf16(&node.value, self.cfg.ascii_only, target);
+        let mut value = get_quoted_utf16(&node.value, self.cfg.ascii_only, target);
+
+        if self.cfg.inline_script {
+            let temp = INLINE_SCRIPT_REGEXP.replace_all(&value, "<\\/$1$2");
```

Comment:
This is unconditional, so I think this will hurt performance a lot

Reply:
Can you run `cargo xtask bench --bench full -p swc_ecma_minifier --features concurrent` from the swc_ecma_minifier directory?

Reply:
The outcome obtained on my local setup (macOS with M1 Max and 32GB RAM) is as follows:

```log
es/minify/libraries/antd
                        time:   [674.34 ms 688.30 ms 702.69 ms]
                        change: [+0.3677% +2.7301% +5.0933%] (p = 0.06 > 0.05)
                        No change in performance detected.
es/minify/libraries/d3  time:   [156.08 ms 157.07 ms 158.67 ms]
                        change: [-4.5544% -1.6954% +0.9294%] (p = 0.29 > 0.05)
                        No change in performance detected.
Found 2 outliers among 10 measurements (20.00%)
  1 (10.00%) low mild
  1 (10.00%) high severe
es/minify/libraries/echarts
                        time:   [567.49 ms 575.56 ms 584.47 ms]
                        change: [-1.7418% +0.2185% +2.2958%] (p = 0.85 > 0.05)
                        No change in performance detected.
es/minify/libraries/jquery
                        time:   [46.225 ms 46.792 ms 48.073 ms]
                        change: [-3.3368% +0.0404% +4.1184%] (p = 0.99 > 0.05)
                        No change in performance detected.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high severe
es/minify/libraries/lodash
                        time:   [58.824 ms 58.984 ms 59.236 ms]
                        change: [-2.5918% -0.8058% +0.8934%] (p = 0.42 > 0.05)
                        No change in performance detected.
es/minify/libraries/moment
                        time:   [28.082 ms 29.361 ms 31.371 ms]
                        change: [-2.3957% +2.0764% +7.4944%] (p = 0.49 > 0.05)
                        No change in performance detected.
Found 2 outliers among 10 measurements (20.00%)
  2 (20.00%) high severe
es/minify/libraries/react
                        time:   [10.016 ms 10.041 ms 10.063 ms]
                        change: [-13.674% -7.7346% -2.1842%] (p = 0.02 < 0.05)
                        Performance has improved.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high mild
es/minify/libraries/terser
                        time:   [121.83 ms 123.95 ms 127.60 ms]
                        change: [-30.179% -24.337% -17.321%] (p = 0.00 < 0.05)
                        Performance has improved.
es/minify/libraries/three
                        time:   [205.10 ms 211.87 ms 223.00 ms]
                        change: [-2.4539% +1.6492% +7.2617%] (p = 0.64 > 0.05)
                        No change in performance detected.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high severe
es/minify/libraries/typescript
                        time:   [1.2369 s 1.2629 s 1.2959 s]
                        change: [-2.5547% +0.6530% +4.0843%] (p = 0.72 > 0.05)
                        No change in performance detected.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high mild
es/minify/libraries/victory
                        time:   [300.70 ms 304.42 ms 308.60 ms]
                        change: [-0.2889% +1.0099% +2.5316%] (p = 0.22 > 0.05)
                        No change in performance detected.
es/minify/libraries/vue time:   [69.470 ms 69.837 ms 70.272 ms]
                        change: [-1.7902% -0.6339% +0.4314%] (p = 0.29 > 0.05)
                        No change in performance detected.
Found 2 outliers among 10 measurements (20.00%)
  1 (10.00%) low mild
  1 (10.00%) high mild


```

Reply:
Can you run it on main and compare?

Reply:
t does appear to be a slight regression. I am curious to see if it could impact performance even when ﻿`self.cfg.inline_script` is set to false.

```log
es/minify/libraries/antd
                        time:   [656.83 ms 674.49 ms 697.47 ms]
                        change: [-5.3373% -2.0059% +1.6161%] (p = 0.31 > 0.05)
                        No change in performance detected.
Found 2 outliers among 10 measurements (20.00%)
  2 (20.00%) high severe
es/minify/libraries/d3  time:   [156.85 ms 157.32 ms 157.83 ms]
                        change: [-1.8192% -0.4254% +0.6690%] (p = 0.59 > 0.05)
                        No change in performance detected.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high mild
es/minify/libraries/echarts
                        time:   [551.56 ms 554.86 ms 558.26 ms]
                        change: [-5.1854% -3.5968% -2.0948%] (p = 0.00 < 0.05)
                        Performance has improved.
es/minify/libraries/jquery
                        time:   [46.467 ms 46.705 ms 46.943 ms]
                        change: [-5.2028% -1.3148% +1.3721%] (p = 0.60 > 0.05)
                        No change in performance detected.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high mild
es/minify/libraries/lodash
                        time:   [58.672 ms 59.072 ms 59.555 ms]
                        change: [-0.6884% +0.0916% +0.9012%] (p = 0.82 > 0.05)
                        No change in performance detected.
es/minify/libraries/moment
                        time:   [27.963 ms 28.081 ms 28.189 ms]
                        change: [-8.0142% -3.3947% +0.6118%] (p = 0.19 > 0.05)
                        No change in performance detected.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high mild
es/minify/libraries/react
                        time:   [9.9995 ms 10.037 ms 10.078 ms]
                        change: [-0.6061% -0.1557% +0.3612%] (p = 0.54 > 0.05)
                        No change in performance detected.
Found 1 outliers among 10 measurements (10.00%)
  1 (10.00%) high mild
es/minify/libraries/terser
                        time:   [121.52 ms 121.89 ms 122.24 ms]
                        change: [-4.7444% -2.7476% -0.9201%] (p = 0.01 < 0.05)
                        Change within noise threshold.
es/minify/libraries/three
                        time:   [205.12 ms 206.19 ms 207.33 ms]
                        change: [-7.5880% -2.6854% +0.6446%] (p = 0.36 > 0.05)
                        No change in performance detected.
es/minify/libraries/typescript
                        time:   [1.2035 s 1.2108 s 1.2187 s]
                        change: [-6.6409% -4.1225% -1.9939%] (p = 0.00 < 0.05)
                        Performance has improved.
es/minify/libraries/victory
                        time:   [296.30 ms 297.94 ms 299.62 ms]
                        change: [-3.5582% -2.1308% -0.7599%] (p = 0.01 < 0.05)
                        Change within noise threshold.
es/minify/libraries/vue time:   [69.824 ms 70.070 ms 70.313 ms]
                        change: [-0.5581% +0.2964% +1.1520%] (p = 0.53 > 0.05)
                        No change in performance detected.
```

Reply:
Upon re-running the benchmark, it appears that the fluctuations observed earlier were mere noise (as indicated in the second comment).

Reply:
If ` ﻿self.cfg.inline_script`, it will not affect the performance but the problem is that `inline_script` is true by default for the minifier

Reply:
> problem is that inline_script is true by default for the minifier

let's represent this using a state machine.